src/domains/ecommerce-common/03_storage.tf

```
@@ -73,50 +73,101 @@ module "ecommerce_storage" {
   name                 = "${local.project}-transactions-expiration-queue"
-  storage_account_name = module.ecommerce_storage.name
+  storage_account_name = module.ecommerce_storage_transient.name
 }
 
 resource "azurerm_storage_queue" "transactions_close_payment_queue" {
   name                 = "${local.project}-transactions-close-payment-queue"
-  storage_account_name = module.ecommerce_storage.name
+  storage_account_name = module.ecommerce_storage_transient.name
 }
 
 resource "azurerm_storage_queue" "transactions_close_payment_retry_queue" {
   name                 = "${local.project}-transactions-close-payment-retry-queue"
-  storage_account_name = module.ecommerce_storage.name
+  storage_account_name = module.ecommerce_storage_transient.name
 }
 
 resource "azurerm_storage_queue" "transactions_refund_retry_queue" {
   name                 = "${local.project}-transactions-refund-retry-queue"
-  storage_account_name = module.ecommerce_storage.name
+  storage_account_name = module.ecommerce_storage_transient.name
 }
 
 resource "azurerm_storage_queue" "transactions_refund_queue" {
   name                 = "${local.project}-transactions-refund-queue"
-  storage_account_name = module.ecommerce_storage.name
+  storage_account_name = module.ecommerce_storage_transient.name
 }
 
 resource "azurerm_storage_queue" "transactions_notifications_retry_queue" {
   name                 = "${local.project}-transaction-notifications-retry-queue"
-  storage_account_name = module.ecommerce_storage.name
+  storage_account_name = module.ecommerce_storage_transient.name
 }
 
 resource "azurerm_storage_queue" "transactions_notifications_queue" {
   name                 = "${local.project}-transaction-notifications-queue"
-  storage_account_name = module.ecommerce_storage.name
+  storage_account_name = module.ecommerce_storage_transient.name
+}
+
+resource "azurerm_private_endpoint" "storage_deadletter_private_endpoint" {
+  count = var.env_short != "d" ? 1 : 0
+
+  name                = "${local.project}-storage-private-endpoint"
+  location            = var.location
+  resource_group_name = azurerm_resource_group.storage_ecommerce_rg.name
+  subnet_id           = module.ecommerce_storage_snet.id
+
+  private_dns_zone_group {
+    name                 = "${local.project}-storage-private-dns-zone-group"
+    private_dns_zone_ids = [data.azurerm_private_dns_zone.storage.id]
+  }
+
+  private_service_connection {
+    name                           = "${local.project}-storage-private-service-connection"
+    private_connection_resource_id = module.ecommerce_storage_deadletter.id
+    is_manual_connection           = false
+    subresource_names              = ["queue"]
+  }
+
+  tags = var.tags
+}
+
+module "ecommerce_storage_deadletter" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//storage_account?ref=v6.7.0"
+
+
+  name                            = replace("${local.project}-sa", "-", "")
```

This has the same name as the `transient` one :)

```suggestion
  name                            = replace("${local.project}-deadletter-sa", "-", "")
```